src/ast/desugar.rs

```diff
@@ -44,17 +44,59 @@ pub(crate) fn desugar_command(
             name,
             variants,
         } => desugar_datatype(span, name, variants),
-        Command::Rewrite(ruleset, rewrite, subsume) => desugar_rewrite(
-            ruleset,
-            rewrite.to_string().replace('\"', "'").into(),
-            &rewrite,
-            subsume,
-        ),
-        Command::BiRewrite(ruleset, rewrite) => desugar_birewrite(
-            ruleset,
-            rewrite.to_string().replace('\"', "'").into(),
-            &rewrite,
-        ),
+        Command::Datatypes { span: _, datatypes } => {
+            let mut res = vec![];
```

Comment:
```suggestion
            // first declare all the datatypes as sorts, then add all explicit sorts which could refer to the datatypes, and finally add all the variants as functions
            let mut res = vec![];
```

Reply:
Thought a little comment here might be useful to help follow the desugaring logic

Reply:
Thank you!